packages/core/src/internal/io.js

```diff
@@ -112,34 +112,52 @@ export function spawn(fn, ...args) {
   return detach(fork(fn, ...args))
 }
 
-export function join(...tasks) {
-  if (tasks.length > 1) {
-    return all(tasks.map(t => join(t)))
-  }
-
-  const task = tasks[0]
+const printJoinDeprecationWarning = once(() =>
+  log('warn', '`join(...tasks)` is deprecated. Please use `join([...tasks])` to join multiple tasks.'),
+)
 
+export function join(taskOrTasks) {
+  if (arguments.length > 1) {
```

Comment:
IMHO we are at perfect place for making breaking changes, let's just drop support for `join(...tasks)` & `cancel(...tasks)`

contextual breaking change warning is nice though! we could move that env check into `if (arguments.length > 1)` & drop `taskOrTasks = array.from(arguments)` (and ofc change the deprecation warning to throw error so the developers can quickly fix their call sites)

Reply:
👍